### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by nilab-khwaja**
Create a class called Solver.
Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.
Create a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it returns "olleh").
Create a method called fizzbuzz that takes one argument, an integer N, and returns a string. The returned string is constructed following these rules:
When N is divisible by 3, return "fizz".
When N is divisible by 5, return "buzz".
When N is divisible by 3 and 5, return "fizzbuzz".
Any other case, return N as a string (e.g. say N is 7 then return "7").

### CODE REVIEW

**Review by omar-muhamad — COMMENTED** [review]
# Invalid ❌

Hi @nilab-khwaja and @Izebiafe, 

#### While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

#### Your Code Review Request is marked as invalid for this reason:

- [ ] Some linters checks are not passed. Please make sure that all checks are passing and green.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1 +1,173 @@
...
+I would like to thank Microverse for giving me the opportunity to learn and grow as a developer.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## 📝 License <a name="license"></a>
+
+This project is [MIT] licensed.
+
```
Comment (Gambit142) [inline README.md]: - [ ] The link to your **_MIT LICENSE_** was omitted in your **_README_**. It is important to include the **_MIT LICENSE_** link in the **_README_** of a repository to clearly communicate the terms under which the code is shared and to establish legal permissions and limitations for potential users and contributors. 😃 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @nilab-khwaja @Izebiafe 👋, 

Good job so far!

## Highlights
- Setup linters correctly ✔️ 
- The team understands the concept of TDD ✔️ 
- Created `Solver` class afterwards ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by BenMukebo — APPROVED** [review]
### Hi @nilab-khwaja, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**